components/grade-table.js

```diff
@@ -1,21 +1,48 @@
 class GradeTable {
-  constructor(tableElement) {
-    this.tableElement = tableElement
+  constructor(tableElement, noGradesElement) {
+    this.tableElement = tableElement;
+    this.noGradesElement = noGradesElement;
+
   }
   updateGrades(grades) {
     var tbody = document.querySelector('tbody');
-    tbody.textContent = ''
+    tbody.innerHTML = ''
     for (var i = 0; i < grades.length; i++) {
-      var tr = document.createElement("tr")
-      var tdname = document.createElement("td")
-      var tdcourse = document.createElement("td")
-      var tdgrade = document.createElement("td")
-
-      tdname.textContent = grades[i].name;
-      tdcourse.textContent = grades[i].course;
-      tdgrade.textContent = grades[i].grade;
-      tr.append(tdname, tdcourse, tdgrade);
-      tbody.append(tr)
+      var row = this.renderGradeRow(grades[i], this.deleteGrade);
+      tbody.append(row)
     }
+var p = document.querySelector('p');
```

Comment:
Use the reference you saved in the `constructor` instead of re-querying the whole `document`.

Reply:
This is still an issue